### PR TITLE
Create Student at Server

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added users DTO
- Created route for student registration in the controller
- Added functionality at user service and repository
- Added constructor at entity files

## Associated Ticket Link(s)

- https://trello.com/c/0eojsIHB

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: src/users/users.repository.ts [hunk 1]
```diff
@@ -2,7 +2,42 @@ import { Injectable } from "@nestjs/common";
...
+
+    const user = new User(firstName, lastName, gender, address, phoneNumber, email, password);
+    const student = new Student(
+      educationLevel,
+      medium,
+      className,
+      degree,
+      degreeName,
+      semesterYear,
+    );
```
Comment (ishmam-iqbal-sazim) [inline src/users/users.repository.ts:33]: You should not need to do it this way. You should structure the input object to match the entity structure.
These should simply be 
`this.usersRepository.createOne(userObject)`
`this.studentsRepository.createOne(studentObject)`

File: src/users/users.repository.ts [hunk 2]
```diff
@@ -2,7 +2,42 @@ import { Injectable } from "@nestjs/common";
...
+
+    user.role = EUserRole.STUDENT;
+    user.student = student;
+    student.user = user;
+
+    this.em.persist([user, student]);
+
+    return user;
```
Comment (ishmam-iqbal-sazim) [inline src/users/users.repository.ts:41]: This should be a transaction. 
DB should rollback if one or the other db action fails. Look into [transactions from mikroORM docs](https://mikro-orm.io/docs/transactions)

File: src/users/users.dtos.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  updatedAt!: string;
+  studentInfo!: StudentResponse;
+}
+
+export class StudentResponse {
+  id!: number;
+  createdAt!: string;
+  updatedAt!: string;
+  educationLevel!: string;
+}
```
Comment (ishmam-iqbal-sazim) [inline src/users/users.dtos.ts:78]: Where are we using these classes?

Comment (faysal-alam-sazim) [reply src/users/users.dtos.ts:78]: > Where are we using these classes?

Response DTO after creating user. Might change it later.

Comment (ishmam-iqbal-sazim) [reply src/users/users.dtos.ts:78]: Please delete any  classes that are not being used. Create as per necessity.

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment.

**Pull request merged**